Adding an entry for Zookeeper (#5951)
istio/istio.io: content/en/faq/applications/zookeeper.md

---
title: Can I run Zookeeper inside an Istio mesh?
description: How to run Zookeeper with Istio.
weight: 50
keywords: [zookeeper]
---

By default, Zookeeper listens on the pod IP address for communication
between servers. Istio and other service meshes require `localhost`
(`127.0.0.1`) to be the address to listen on.

There is a configuration parameter that can be used to change this
default behavior:

> [`quorumListenOnAllIPs`](https://zookeeper.apache.org/doc/r3.5.6/zookeeperAdmin.html).

Reviewer: Since this is option is listed in the doc under `Unsafe Options` and also because it isn't just changing the listen IP to `localhost` as the above says is needed, I think you should add another sentence to explain what it does and why it is not unsafe to listen on all IPs in this case.

Author: yes, I am trying to find out why this maybe unsafe. Asking on the zookeeper mailing list as there is no explanation in the admins guide.

Author: @frankbu I have not received any response on the Zookeeper mailing list but added the requested explanation and noted that the option is under "unsafe options". I do not see a particular issue with listening on all addresses and there are a lot of apps that do just that. We can update this entry if/when we get more info.

Reviewer: I wouldn't mention anything about it being an unsafe option if you don't also explain why it's not a problem in this case.

Set this parameter to `true` by using the following command where

> `$ZK_CONFIG_FILE` is the Zookeeper configuration file.

Reviewer: ```suggestion
`$ZK_CONFIG_FILE` is your Zookeeper configuration file.
```

Author: Done.

---
title: Can I run Zookeeper inside an Istio mesh?
description: How to run Zookeeper with Istio.
weight: 50
keywords: [zookeeper]
---

By default, Zookeeper listens on the pod IP address for communication
between servers. Istio and other service meshes require `localhost`
(`127.0.0.1`) to be the address to listen on.

There is a configuration parameter that can be used to change this
default behavior:
[`quorumListenOnAllIPs`](https://zookeeper.apache.org/doc/r3.5.6/zookeeperAdmin.html).

> This option allows Zookeeper to listen on all addresses including the

Reviewer: ```suggestion
This option instructs Zookeeper to listen on all addresses including `localhost` as required by Istio.
```

> `localhost`. It should be noted that this option is listed under the

Reviewer: ```suggestion
```

> `Unsafe Options` category. Set this parameter to `true` by using the

Reviewer: ```suggestion
Set this parameter to `true` by using the
```
